vars/updatecli.groovy

```diff
@@ -0,0 +1,34 @@
+
+def call(userConfig = [:]) {
+  def defaultConfig = [
+    action: 'diff',
+    config: './updatecli/updatecli.d',
+    values: './updatecli/values.yaml',
+    updatecliDockerImage: 'ghcr.io/updatecli/updatecli:latest',
+  ]
+
+  // No Deep Merge - https://e.printstacktrace.blog/how-to-merge-two-maps-in-groovy/
+  final Map finalConfig = defaultConfig + userConfig
```

Comment:
not sure if any advantage, but we normally use this shorter form: https://github.com/hmcts/cnp-jenkins-library/blob/master/vars/notifyBuildEvent.groovy#L19-L22

Reply:
Thanks @timja ! I have to admit that, even though I understand with your comment that this code is somehow merging the 2 maps, I'm not sure to really understand the operator `<<` . The Groovy documentation at https://groovy-lang.org/operators.html#_bit_shift_operators tells me that it's a shift operator, but how does it work with a Map ? Is there an override in the Map class or something?

Reply:
it's called `leftShift()` https://blog.mrhaki.com/2010/04/groovy-goodness-adding-maps-to-map_21.html

Reply:
@timja Switched to your proposal, with a link to the page you mentioned. Thanks!